evm_loader/program/src/executor_state.rs

```diff
@@ -173,10 +173,10 @@ impl ExecutorSubstate {
         let mut entering = Self {
             metadata: self.metadata.spit_child(gas_limit, is_static),
             parent: None,
-            logs: Vec::new(),
-            accounts: BTreeMap::new(),
-            storages: BTreeMap::new(),
-            deletes: BTreeSet::new(),
+            logs: self.logs.clone(),
+            accounts: self.accounts.clone(),
+            storages: self.storages.clone(),
+            deletes: self.deletes.clone(),
         };
```

Comment:
It is a bad idea. Cloning there will cause an exponential increase in the instructions count for iterative execution. As well as storage account size.

See also how it is done in the original EVM - https://github.com/rust-blockchain/evm/blob/2a8a3e967ec265fdc7120ff6b57ceab021ca69f6/src/executor/stack/state.rs#L100